packages/ui/src/views/repo/repo-commits/components/repo-commits-branch-selector.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  Text
+} from '@/components'
+import { cn } from '@utils/cn'
+
+export interface BranchListProps {
+  name?: string
+}
+
+interface PageProps {
+  name: string
+  branchList?: BranchListProps[]
+  size?: 'default' | 'sm'
+  width?: 'auto' | 'sm' | 'md' | 'lg' | 'full'
+  selectBranch: (branch: string) => void
+  prefix?: string
+}
+
+const RepoCommitsBranchSelector = ({ ...props }: PageProps) => {
```

Comment:
we already have a branch selector component on RepoSummary page. let's re-use that instead of adding a copy specific to repocommits page.